app_test.go

```diff
@@ -2340,6 +2342,28 @@ func TestDone(t *testing.T) {
 	}
 }
 
+// TestShutdownThenWait tests that if we call .Shutdown before waiting, the wait
+// will still return the last shutdown signal.
+func TestShutdownThenWait(t *testing.T) {
+	t.Parallel()
+
+	var (
+		s Shutdowner
+	)
+	app := fxtest.New(
+		t,
+		Populate(&s),
+	)
+	defer app.RequireStop()
+	assert.NotNil(t, s)
+
+	err := s.Shutdown(ExitCode(1337))
+	require.NoError(t, err)
+
+	shutdownSig := <-app.Wait()
```

Comment:
Because of GitHub workflow, I cannot comment the Wait method, so I'm commenting here.

I would expect to use the context as argument

This way `Wait` could stop when ctx is done.
The same way  `Stop` does.

It's useless in the unit tests, but meaningful in the real app

Reply:
I'm unsure but I would expect the same from `app.Done`

Reply:
Last remark, I'm surprised that code in Stop are not closing the channels when leaving on ctx.Done()